app/src/main/java/com/example/habitsmasher/ui/home/HomeFragment.java

```diff
@@ -1,66 +1,273 @@
-        Habit testHabit = new Habit("This is a title", "This is my reason", new Date(), "MO",
-                true, DatabaseEntity.generateId(), new HabitEventList());
+        TextView date = view.findViewById(R.id.home_date);
+        date.setText(getDateString());
 
-        // Getting the corresponding textviews in the fragment_home.xml
-        TextView titleView = binding.habitTitle;
-        TextView reasonView = binding.habitReason;
-        TextView dateView = binding.habitDate;
+        // query firebase for all habits that correspond to the current user
+        Query query = getListFromFirebase();
 
-        // Setting the pattern for the date
-        String pattern = "MM-dd-yyyy";
-        SimpleDateFormat simpleDateFormat = new SimpleDateFormat(pattern, Locale.CANADA);
+        // populate the list with existing items in the database
+        FirestoreRecyclerOptions<Habit> options = new FirestoreRecyclerOptions.Builder<Habit>()
+                .setQuery(query, Habit.class)
+                .build();
 
-        // Updating the text fields of the textviews
-        titleView.setText(testHabit.getTitle());
-        reasonView.setText(testHabit.getReason());
-        dateView.setText(simpleDateFormat.format(testHabit.getDate()));
+        populateList(query);
+        _habitItemAdapter = new HomeHabitItemAdapter(options, _habitList, _user.getUsername());
+        LinearLayoutManager layoutManager = new LinearLayoutManager(_context,
+                LinearLayoutManager.VERTICAL,
+                false);
 
+        //View view = inflater.inflate(R.layout.fragment_habit_list, container, false);
+        initializeRecyclerView(layoutManager, view);
 
-        return root;
+
+        return view;
     }
 
     @Override
     public void onDestroyView() {
         super.onDestroyView();
-        binding = null;
+    }
+
+    @Override
+    public void onStart() {
+        super.onStart();
+        _habitItemAdapter.startListening();
+    }
+
+    @Override
+    public void onStop() {
+        super.onStop();
+        _habitItemAdapter.stopListening();
+    }
+
+    @NonNull
+    private User getCurrentUser() {
+        SharedPreferences sharedPref = _context.getSharedPreferences(USER_DATA_PREFERENCES_TAG, Context.MODE_PRIVATE);
+
+        String username = sharedPref.getString("username", "user");
+        String userId = sharedPref.getString("userId", "id");
+        String email = sharedPref.getString("email", "email");
+        String password = sharedPref.getString("password", "password");
+
+        return new User(userId, username, email, password);
+    }
+
+    @Override
+    protected Query getListFromFirebase() {
+
+        // returns all of the habits that take place on the same day (Today)
+        return _db.collection("Users")
+                .document(_user.getId())
+                .collection("Habits")
+                .whereArrayContains("days", getCurrentDay());
+    }
```

Comment:
Gets all the habits that take place on the current day. This is what all of the DaysTracker changes lead up to.